### PR TITLE
WIP: Feature/various comments

### PR DESCRIPTION
**Pull request by esiqveland**
Not sure how you take these kind of suggestions =)

This PR will be my work in progress to try to get into the project codebase.

My first goal was being able to add some tests, but as it is, this is not overly testable, so a lot of refactoring have to go down...

Please feel free to comment if you like or don't like something. Suggestions very welcome!

### CODE REVIEW

**Comment by Veritasimo** [comment]
Apologies for that. The server was developed over a rather rushed 2-4 week period in parallel with the C++ library. It was initially meant to be my own personal prototype. :D

**Comment by esiqveland** [comment]
Haha, well it works quite well tbh! =)

There is a lot of globally defined state floating around though, so it's hard to write tests as is.

**Comment by Veritasimo** [comment]
IIRC adding websocket support wasn't that much of difficult refactor. You could probably get ClientConnection working with a mock socket quite easily.

File: chat.go [hunk 1]
```diff
@@ -80,6 +86,7 @@ func GetChatRoom(name string, password string, joinable, fixed bool) (room *Chat
 	return room
 }
 
+// run is the main loop that runs and handles events for a ChatRoom.
 func (cr *ChatRoom) run() {
 	defer func() {
```
Comment (esiqveland) [inline chat.go]: Could we do this in the Close() function and just defer cr.Close()?

Comment (Veritasimo) [inline chat.go]: You could, although if memory serves it's probably a bit redundant to have all of these functions public as `ChatRoom`s are essentially an isolated environment aside from the channels used to add and remove users.

Comment (esiqveland) [inline chat.go]: Yeah, they should probably be private. It was more of a self-reminder which ones are helper functions and which ones should actually be called.

I will put this in the close func then.

**Pull request merged**